oak_functions/logger/src/lib.rs

```diff
@@ -0,0 +1,32 @@
+//
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use log::Level;
+
+pub trait OakLogger {
+    /// Logs the message, which might contain sensitive information, at the specified `Level`.
+    ///
+    /// Only insecure debug-only implementations may provide a non-empty implementation. Production
+    /// implementations must not do anything.
+    #[allow(unused_variables)]
```

Comment:
I would avoid the default methods unless necessary, IMO it makes this look too much OOP, while it's not really. I think it should be fine to just always implement them explicitly?

Reply:
Removed. I added it only to make clear that it should be an empty implementation by default.